tests/E2E Tests/GraphServiceClientTests/GraphServiceClientTests.cs

```diff
@@ -67,5 +72,45 @@ async Task Test()
             });
 
         }
+
+        [Fact]
+        public async Task AuthenticateRequestAsync_NonGraphUri_DoesNotSetAuthZHeader()
+        {
+            // arrange
+            RequestInformation request = new()
+            {
+                URI = new Uri("http://www.contoso.com/")
+            };
+
+            GraphAuthenticationProvider graphAuthenticationProvider = new(_authorizationHeaderProvider, new GraphServiceClientOptions());
+
+            // act
+            await graphAuthenticationProvider.AuthenticateRequestAsync(request).ConfigureAwait(false);
+
+            // assert
+            Assert.False(request.Headers.ContainsKey("Authorization"));
+        }
+
+        [Fact]
+        public async Task AuthenticateRequestAsync_GraphUri_SetsAuthZHeader()
+        {
+            // arrange
+            RequestInformation request = new()
+            {
+                URI = new Uri("https://graph.microsoft.com/.default")
+            };
+
+            var mockAuthorizationHeaderProvider = new Mock<IAuthorizationHeaderProvider>();
```

Comment:
Tried to setup an IAuthorizationHeaderProvider like so: 

```csharp
TokenAcquirerFactory tokenAcquirerFactory = TokenAcquirerFactory.GetDefaultInstance();
IServiceCollection services = tokenAcquirerFactory.Services;

services.Configure<MicrosoftIdentityApplicationOptions>(s_optionName, option =>
    {
        option.Instance = "https://login.microsoftonline.com/";
        option.TenantId = "msidlab4.onmicrosoft.com";
        option.ClientId = "f6b698c0-140c-448f-8155-4aa9bf77ceba";
    });
IServiceProvider serviceProvider = tokenAcquirerFactory.Build();
IAuthorizationHeaderProvider authorizationHeaderProvider = serviceProvider.GetRequiredService<IAuthorizationHeaderProvider>();         
```
However, couldn't seem to get it to work. Set up a mock for now to show that the code works as intended and that an authorization header is appropriately added when RequestInformation.URI is a graph URI.

Reply:
Discussed offline with @jmprieur - removing as we have plenty of positive tests for the graph service client